Content/Items/Haunted/Weapons.EchochainWhip.cs

```diff
@@ -0,0 +1,703 @@
+			Projectile.width = 10;
+			Projectile.height = 10;
+			Projectile.friendly = true;
+			Projectile.tileCollide = false;
+			Projectile.ignoreWater = true;
+		}
+
+		public override bool? CanDamage()
+		{
+			return false;
+		}
+
+		public override bool PreAI()
+		{
+			Projectile.Center = DeathTimer > 0 ? oldMouse : OwnerMouse;
+			PopulateTilesAndTargets();
+
+			return true;
+		}
+
+		public override void AI()
+		{
+			if (DeathTimer > 0)
+			{
+				float progress = 1f - DeathTimer / 30f;
+
+				Projectile.timeLeft = 2;
+				DeathTimer--;
+
+				if (DeathTimer == 1)
+				{
+					Projectile.Kill();
+					Owner.itemTime = 0;
+					Owner.itemAnimation = 0;
+					Owner.SetCompositeArmFront(true, Player.CompositeArmStretchAmount.Full, 0f); // gotta set the composite arm here again otherwise the players front arm appears out for a frame and it looks bad		
+					return;
+				}
+
+				if (progress < 0.25f)
+				{
+					handRotation = MathHelper.Lerp(-1f, -3f, EaseBuilder.EaseCircularInOut.Ease(1f - (DeathTimer - 22.5f) / 7.5f)) * Owner.direction;
+				}
+				else
+				{
+					handRotation = MathHelper.Lerp(-3f, -0.5f, EaseBuilder.EaseQuarticIn.Ease(1f - DeathTimer / 22.5f)) * Owner.direction;
+					if (progress >= 0.9f && !playedSound)
+					{
+						Helper.PlayPitched("Effects/HeavyWhooshShort", 1f, 0f, Owner.Center);
+
+						CameraSystem.shake += 6;
+						playedSound = true;
+					}
+				}
+			}
+			else
+			{
+				Projectile.rotation = Owner.DirectionTo(OwnerMouse).ToRotation();
+
+				if (!CanHold)
+				{				
+					Projectile.timeLeft = 5;
+					DeathTimer = 30f;
+					oldMouse = OwnerMouse;
+					CameraSystem.shake += 4;
+					Helper.PlayPitched("Effects/HeavyWhoosh", 1f, 0f, Owner.Center);
+
+					for (int i = 0; i < Main.rand.Next(2, 4); i++)
+					{
+						Dust.NewDustPerfect(Owner.Center + new Vector2(15f * Owner.direction, -5f) + Main.rand.NextVector2CircularEdge(5f, 5f), ModContent.DustType<Dusts.GlowLine>(), -Vector2.UnitY.RotatedByRandom(0.2f) * 1.5f, 200, new Color(130, 255, 50, 0), 0.35f);
+					}
+
+					var targetsToChain = new List<NPC>();
+
+					for (int i = 0; i < enemyTiles.Length; i++)
+					{
+						if (enemyTiles[i] != null && targets[i] != null && targets[i].active && Collision.CanHitLine(new Vector2(enemyTiles[i].Value.X * 16, enemyTiles[i].Value.Y * 16) + new Vector2(0f, -30f), 2, 2, targets[i].Center, 2, 2))
+						{
+							Vector2 pos = new Vector2(enemyTiles[i].Value.X * 16, enemyTiles[i].Value.Y * 16);
+
+							targetsToChain.Add(targets[i]);
+
+							int[] frames = new int[17];
+							for (int a = 0; a < frames.Length; a++) // populate array
+							{
+								frames[a] = Main.rand.Next(3);
+							}
+
+							var proj = Projectile.NewProjectileDirect(null, pos, Vector2.Zero, ModContent.ProjectileType<EchochainWhipAltProjectileChain>(), Projectile.damage, Projectile.knockBack, Projectile.owner);
+
+							EchochainWhipAltProjectileChain chain = proj.ModProjectile as EchochainWhipAltProjectileChain;
```

Comment:
using the output of the Projectile.newProjectile is pretty much always a trap when it comes to multiplayer. this will not get synced normally and I would recommend following the ToAssign paradigm like the laser pointer in #611. make sure you remember to send the extra fields too.